docs/ruleset_functions/get_spaces_associated_with_each_SWH_bat.md

```diff
@@ -0,0 +1,64 @@
+                    - calculate the total hot water used using the function get_energy_required_to_heat_swh_use and add it to the shw_use_dict: `swh_use_dict[space.service_water_heating_area_type] += get_energy_required_to_heat_swh_use(swh_use)`
+            - otherwise, need to look at the individual swh uses: `else:`
+                - look at each swh use for the sapce: `for swh_use in space.service_water_heating_uses:`
+                    - calculate the total hot water used using the function get_energy_required_to_heat_swh_use: `total_energy_used = get_energy_required_to_heat_swh_use(swh_use)`
+                    - if the shw_use has an area type, add it to the swh_use_dict: `if swh_use.area_type:`
+                        - add the shw_use area type to the swh_use_dict, if it doesn't exist yet: `swh_use_dict.set_default(swh_use.area_type, 0)`
+                        - add the total energy used to the dict: `swg_use_dict[swh_use.area_type] += get_energy_required_to_heat_swh_use(swh_use)`
+                    - otherwise: `else:`
+                        - add the energy use to type "UNDETERMINED": `swh_use_dict.set_default("UNDETERMINED", 0)`
+                        - add the total energy used to the dict: `swh_use_dict["UNDETERMINED"] += get_energy_required_to_heat_swh_use(swh_use)`
+        - now we need to determine the building_segment swh_use_type based on the following rules:
+        -     1. At least 50% of the SWH uses needs to be assigned a use type
+        -     2. All SWH needs to be assigned to the same use type
+        - if there is only one element in swh_use_dict, the building_segment_shw_bat is the only key in swh_use_dict: `if(len(swh_use_dict)==1): building_segment_shw_bat = list(swh_use_dict.keys())[0]`
+        - otherwise, if swh_use_dict has two elements: `elsif(len(swh_use_dict)) == 2:`
+            - this should be "UNDETERMINED" and one other area use type: `if "UNDETERMINED" in swh_use_dict:`
+                - get the other key.  First get a list of all keys: `all_keys = list(swh_use_dict.keys())`
+                - remove "UNDETERMINED": `all_keys.remove("UNDETERMINED")`
+                - set other_key to the remaining item in the list: `other_key = all_keys[0]`
+                - now check whether the water use for UNDETERMINED is < the water use for the other key: `if swh_use_dict["UNDETERMINED"] < swh_use_dict[other_key]:`
+                    - set the building_segment_shw_bat equal to other_key: `building_segment_shw_bat = other_key`
+    - create the list in shw_and_spaces_dict for this bat if it doesn't exist: `shw_and_spaces_dict.set_default(building_segment.service_water_heating_area_type, [])`
+    - look at each space in the building segment: `for space in building_segment.spaces:`
+        - append the space.id to the dictionary: `shw_and_spaces_dict[(building_segment.service_water_heating_area_type].append(space.id)
+
+
+**Returns** shw_and_spaces_dict
```

Comment:
What does this return if the BAT for the segment could not be determine from any of the methods?

Reply:
it'd still be shw_and_spaces_dict, which would be equal to {}